apps/src/javalab/CommitDialog.jsx

```diff
@@ -6,16 +6,50 @@ import StylizedBaseDialog, {
   FooterButton
 } from '@cdo/apps/componentLibrary/StylizedBaseDialog';
 import {connect} from 'react-redux';
+import _ from 'lodash';
+
+const fileShape = PropTypes.shape({
+  name: PropTypes.string.isRequired,
+  commit: PropTypes.bool.isRequired,
+  hasConflictingName: PropTypes.bool.isRequired
+});
 
 export class UnconnectedCommitDialog extends React.Component {
-  state = {
-    filesToBackpack: [],
-    commitNotes: '',
-    backpackSaveInProgress: false,
-    commitSaveInProgress: false,
-    hasBackpackSaveError: false,
-    hasCommitSaveError: false
-  };
+  constructor(props) {
+    super(props);
+
+    let existingBackpackFiles = [];
+    let hasBackpackLoadError = false;
+    if (props.backpackApi.hasBackpack()) {
+      props.backpackApi.getFileList(
+        () => (hasBackpackLoadError = true),
+        filenames => (existingBackpackFiles = filenames)
+      );
+    }
+
+    this.state = {
+      filesToBackpack: [],
+      existingBackpackFiles: existingBackpackFiles,
+      commitNotes: '',
+      backpackSaveInProgress: false,
+      commitSaveInProgress: false,
+      hasBackpackLoadError: hasBackpackLoadError,
+      hasBackpackSaveError: false,
+      hasCommitSaveError: false
+    };
+  }
+
+  // Get updated backpack file list every time we open the modal
+  componentDidUpdate(prevProps) {
+    if (this.props.isOpen !== prevProps.isOpen && this.props.isOpen) {
+      if (this.props.backpackApi.hasBackpack()) {
+        this.props.backpackApi.getFileList(
+          this.handleBackpackLoadError,
+          filenames => this.setState({existingBackpackFiles: filenames})
+        );
+      }
```

Comment:
Might be worth breaking this out into a function since it's used in a couple spots?